src/System.Linq/src/System/Linq/Partition.cs

```diff
@@ -42,22 +42,18 @@ internal interface IPartition<TElement> : IIListProvider<TElement>
 
         IPartition<TElement> Take(int count);
 
-        TElement ElementAt(int index);
+        TElement TryGetElementAt(int index, out bool found);
 
-        TElement ElementAtOrDefault(int index);
+        TElement TryGetFirst(out bool found);
 
-        TElement First();
-
-        TElement FirstOrDefault();
-
-        TElement Last();
-
-        TElement LastOrDefault();
+        TElement TryGetLast(out bool found);
```

Comment:
I'd suspect that it might be affected by whether the variable referenced by the `out` call was a local or in a field (including a local in an iterator), since that has different GC impact. I'm going to try cases that vary in that regard if my Duct Tape Inside™ fix gives me a working dev rig again. Anything else it should vary on?